### ISSUE TITLE
desc files: document/decide on naming standards for elements

### ISSUE THREAD

**Issue by sensestage** [ModalityTeam/Modality-toolkit#102]
document naming standards for elements in ```.desc.scd```-files.
This should be reflected in the documentation (e.g. the how to write a spec tutorial)

**Comment by adcxyz** [comment]
We should mirror all these definitions in a class, e.g. MKtlStandard. 
Then we can automatically generate template files, 
test descriptions for compliance with recommended naming schemes, 
fill in default specs where missing, etc etc.

```
/// first sketch:
MKtlStandard {
    classvar  <deviceTypes, <elemTypes, ...;
    classvar <defaultElemSpecs;
    *initClass {
        deviceTypes = [\faderbox, \mixer, \gamepad];
        elemTypes = (
            \slider: \sl, \button: \bt, knob: \kn, etc
        );
        defaultElemSpecs = (
            midi: (\slider: \midiNum, \button: \midiBut, \knob: \midiNum, etc),
            midi: (\slider: ?, \button: \hidBut, \knob: ?, etc)
        )
    }
}
```

**Comment by LFSaw** [comment]
This should then go to the 

	[ModalityTests](https://github.com/ModalityTeam/ModalityTests)

Quark, yes?

**Comment by adcxyz** [comment]
No, better have them in the toolkit, 
so users can use them for looking up what is required, and which are optional, 
and to check their desc file easily for correctness while writing it.
Will also be useful for templates when making a desc from a captured  input stream, 
and in other places too.

**Comment by LFSaw** [comment]
:+1:

**Comment by adcxyz** [comment]
ok, this seems to be one of the crucial things to finalize next.

**Comment by LFSaw** [comment]
yes.

**Comment by adcxyz** [comment]
updated redef proposal following marijes suggestions in ba725da

**Comment by adcxyz** [comment]
closed with 304842d
